test/priceUtils.js

```diff
@@ -26,5 +33,123 @@ contract("PriceOracle", function(accounts) {
       const price = 1000
       assert(await isPriceReasonable(exchange, targetTokenId, stableTokenId, price, acceptedPriceDeviationInPercentage))
     })
+    it("checks that bracket traders does not sell unprofitable for tokens with the same decimals", async () => {
+      const ERC20 = artifacts.require("DetailedMintableToken")
+      const token1 = await ERC20.new("WETH", 18)
+      const token2 = await ERC20.new("DAI", 18)
+      await prepareTokenRegistration(accounts[0], exchange)
+      await exchange.addToken(token1.address, { from: accounts[0] })
+      await prepareTokenRegistration(accounts[0], exchange)
+      await exchange.addToken(token2.address, { from: accounts[0] })
+
+      const orders = [
+        {
+          // normal order selling for more than 120 DAI per WETH
+          user: "0xf888243aacb5626b520d0028371bad672a477fd8",
+          sellTokenBalance: new BN(0),
+          buyToken: 1,
+          sellToken: 2,
+          priceNumerator: new BN("1").mul(new BN(10).pow(new BN(18))),
+          priceDenominator: new BN("115").mul(new BN(10).pow(new BN(18))),
+        },
+        {
+          // normal order selling for more than 120 DAI per WETH
+          user: "0xf888243aacb5626b520d0028371bad672a477fd8",
+          sellTokenBalance: new BN("125").mul(new BN(10).pow(new BN(18))),
+          buyToken: 2,
+          sellToken: 1,
+          priceNumerator: new BN("132").mul(new BN(10).pow(new BN(18))),
+          priceDenominator: new BN("1").mul(new BN(10).pow(new BN(18))),
+        },
+      ]
+
+      const globalPriceStorage = {}
+      globalPriceStorage["DAI-USDC"] = 1.0
+      globalPriceStorage["WETH-DAI"] = 1 / 120.0
+      globalPriceStorage["WETH-USDC"] = 1 / 120.0
+
+      assert.equal(await checkNoProfitableOffer(orders[0], exchange, globalPriceStorage), true)
+      assert.equal(await checkNoProfitableOffer(orders[1], exchange, globalPriceStorage), true)
+    })
+    it("checks that bracket traders does not sell unprofitable for tokens with the different decimals", async () => {
+      const ERC20 = artifacts.require("DetailedMintableToken")
+      const token1 = await ERC20.new("DAI", 18)
+      const token2 = await ERC20.new("USDC", 6)
+
+      await prepareTokenRegistration(accounts[0], exchange)
+      await exchange.addToken(token1.address, { from: accounts[0] })
+      const DAItokenId = (await exchange.numTokens.call()).toNumber() - 1
+
+      await prepareTokenRegistration(accounts[0], exchange)
+      await exchange.addToken(token2.address, { from: accounts[0] })
+      const USDCtokenId = (await exchange.numTokens.call()).toNumber() - 1
```

Comment:
`addCustomMintableTokenToExchange` could be used also here.

Reply:
Or maybe not? Is `tokenAddressToIdMap` equivalent to `exchange.numTokens - 1` here?